src/main/java/jdbchomework/console/AbstractMenu.java

```diff
@@ -1,21 +1,26 @@
 package jdbchomework.console;
 
 import jdbchomework.MainWithJdbc;
-import org.slf4j.LoggerFactory;
+import org.slf4j.Logger;
 
 import java.io.IOException;
 import java.util.List;
 
+public abstract class AbstractMenu {
 
-public class VisualUserMenu {
+    public static final String CHOOSE_ACTION = "Choose action: ";
+    public static final String ERROR_INCORRECT_MENU_ITEM_SELECTED = "ERROR: Incorrect menu item selected \n";
+    protected MenuContents menuContents;
+    private org.slf4j.Logger log;
 
-    private org.slf4j.Logger log = LoggerFactory.getLogger(VisualUserMenu.class);
-
-    public static void outputSplitLine() {
-        System.out.println("**************************************************************");
+    public AbstractMenu(Logger log, MenuContents menuContents) {
+        this.log = log;
+        this.menuContents = menuContents;
     }
 
-    public static <E> void printListInConsole(List<String> header, List<E> options) {
+    abstract void menu();
+
+    public <E> void printListInConsole(List<String> header, List<E> options) {
         if (header != null) {
             header.forEach(System.out::println);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this usage of System.out or System.err by a logger. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS106)